### PR TITLE
feat(travis): keep staging branch up to date

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
https://github.com/azerothcore/azerothcore-wotlk/wiki/The-Staging-Branch

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
@Rochet2 thanks for your feedback. What do you suggest? I'm not that confident with travis yet.

File: .travis.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ before_install:
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then git fetch --unshallow; fi
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then git checkout $TRAVIS_BRANCH; fi
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then git add -A . && git diff --cached --quiet || git commit -am "Import pending SQL update file" && git push https://$GITHUB_API_KEY@github.com/$TRAVIS_REPO_SLUG.git $TRAVIS_BRANCH; fi
+  # sync staging with master
+  - if [ "$TRAVIS_PULL_REQUEST" = "false" ] && [ "$TRAVIS_BRANCH" = "master" ]; then git checkout staging && git merge --no-edit master && git push https://$GITHUB_API_KEY@github.com/$TRAVIS_REPO_SLUG.git staging; git checkout master; fi
```
Comment (Rochet2) [inline .travis.yml]:
```suggestion
  - if [ "$TRAVIS_PULL_REQUEST" = "false" ] && [ "$TRAVIS_BRANCH" = "master" ] && [ "$GITHUB_API_KEY" = "" ]; then git checkout staging && git merge --no-edit master && git push https://$GITHUB_API_KEY@github.com/$TRAVIS_REPO_SLUG.git staging; git checkout master; fi
```

File: .travis.yml [hunk 2]
```diff
@@ -26,6 +26,8 @@ before_install:
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then git fetch --unshallow; fi
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then git checkout $TRAVIS_BRANCH; fi
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then git add -A . && git diff --cached --quiet || git commit -am "Import pending SQL update file" && git push https://$GITHUB_API_KEY@github.com/$TRAVIS_REPO_SLUG.git $TRAVIS_BRANCH; fi
```
Comment (Rochet2) [inline .travis.yml]:
```suggestion
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ] && [ "$GITHUB_API_KEY" = "" ]; then git add -A . && git diff --cached --quiet || git commit -am "Import pending SQL update file" && git push https://$GITHUB_API_KEY@github.com/$TRAVIS_REPO_SLUG.git $TRAVIS_BRANCH; fi
```

**Comment by Rochet2** [comment]
Did not test, but I think that code should check if `$GITHUB_API_KEY` is not empty. So if not empty, then push.

**Comment by FrancescoBorzi** [comment]
@Rochet2 thanks, I included your suggestion but only in the new part (about the staging branch).

let's first merge this and see if it works, if it does we can fix the sql update import too in another PR.

What do you think?

**Comment by Deku** [comment]
I asked @BarbzYHOOL about the global variables for Travis, but he didn't managed that. Maybe we should document them at the beggining of travis.yml.

**Comment by Rochet2** [comment]
> What do you think?

sure, whatever works

**Comment by FrancescoBorzi** [comment]
@Viste this was supposed to go on master :)

**Comment by BarbzYHOOL** [comment]
i don't have travis access :P (i do'nt even have an account if it's required)

**Pull request merged**